spec/01-lexical-syntax.md

```diff
@@ -6,13 +6,15 @@ chapter: 1
 
 # Lexical Syntax
 
-Scala programs are written using the Unicode Basic Multilingual Plane
-(_BMP_) character set; Unicode supplementary characters are not
-presently supported.  This chapter defines the two modes of Scala's
-lexical syntax, the Scala mode, and the _XML mode_. If not
-otherwise mentioned, the following descriptions of Scala tokens refer
-to _Scala mode_, and literal characters ‘c’ refer to the ASCII fragment
-`\u0000` – `\u007F`.
+Scala source text consists of Unicode characters.
+
+In particular, certain character classes correspond to Unicode
```

Comment:
> certain character classes

Does this refer to the list of classes below?

Does the first half of this paragraph add something to the spec, or only clarify?

Reply:
I added it as clarification in response to sjrd's comment about "character". Unicode does say "character", in our sense as opposed to "what people take as a single character", but OTOH it's inherently complex or confusing. I'm happy to re-edit as many times as necessary; I'm not wedded to any wording.

IMHO, "Scala source text consists of Unicode code points" makes me wonder if they're saying something subtle. But my perception may be warped by writing code. Maybe "Scala supports Unicode source text."

The clarification of "What do you mean, you support Unicode?" is that a math symbol `+` that is not the ascii is taken as an operator, though maybe not the unary one you're thinking of, and that any code point that is a letter can start an identifier, not just BMP. The intention also is to be vague enough not to say something wrong or too much.

Reply:
Another alternative to consider: don't mention that scala supports or is supported by unicode (non)characters and code points, just mention that the spec is defines the scala grammar in terms of unicode:

> The scala lexical grammar is expressed in Unicode code points and Unicode character properties.

Reply:
For the spec I think it's fine to just say "Unicode code points", it's precise and googleable.

Reply:
agree, but I'd also be fine with "Scala source code consists of Unicode text"

Reply:
I'll go with Seth's minimization since he holds the merge button.

Reply:
As I typed it in, I just noticed how Seth cleverly moved "code" to the left and "text" to the right!

Reply:
Also Scala turtles consist of Uniturtle turtles. It's turtles all the way...